toolkit/tools/imagegen/configuration/configuration_test.go

```diff
@@ -413,9 +414,13 @@ var expectedConfiguration Config = Config{
 			KernelOptions: map[string]string{
 				"default": "kernel",
 			},
-			AdditionalFiles: map[string]string{
-				"local/path/file1": "/final/system/path",
-				"local/path/file2": "/final/system/path/renamedfile2",
+			AdditionalFiles: map[string]FileConfigList{
+				"local/path/file1": {{Path: "/final/system/path"}},
+				"local/path/file2": {{Path: "/final/system/path/renamedfile2"}},
+				"local/path/file3": {{Path: "/final/system/path/file3"}},
+				"local/path/file4": {{Path: "/final/system/path/file4", Permissions: ptrutils.PtrTo(FilePermissions(0o664))}},
```

Comment:
I'm open to considering this (`ptrutils.PtrTo`), but maybe in a subsequent PR? If we are going to add this we should switch the existing code. We generally just use `&` in our code to get addresses.

Reply:
Using `&` only works when referring to a named variable or a struct initializer. For things like integers and strings, you have to use a temporary variable, which is annoying. I added `ptrutils.PtrTo` to avoid needing to use temporary variables.

I don't think we need to use `ptrutils.PtrTo` everywhere. Only where `&` doesn't work.

Reply:
ok, fair enough. Creating the dummy variables does get annoying.